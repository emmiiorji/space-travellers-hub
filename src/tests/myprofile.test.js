import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import Myprofile from "../components/myprofile/index";
import setupStore from "./configureStore";

describe("Test rockets in the MyProfile components", () => {
  test("The profile page renders correctly", () => {
    const store = setupStore(false);
    const page = render(
      <Provider store={store}>
        <Myprofile />
      </Provider>
    );
    expect(page).toMatchSnapshot();
  });

  test("Shows the rocket reserved in the profile page", () => {
    const store = setupStore(true);
    const { container } = render(
      <Provider store={store}>
        <Myprofile />
      </Provider>
    );
    expect(container.getElementsByClassName("rocket-container")).toHaveLength(
      1
    );
    expect(screen.queryByText(" No Rockets Reserved Yet")).toBeFalsy();
  });
});
