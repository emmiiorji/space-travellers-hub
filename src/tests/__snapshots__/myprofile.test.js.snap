// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rockets in the MyProfile components The profile page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-container"
      >
        <div
          class="dragon-container"
        >
          <hr
            class="line"
          />
          <div
            class="myprofile-section"
          >
            <div
              class="dragons-section"
            >
              <h3
                class="mp-dragon-title"
              >
                Dragons
              </h3>
              <p
                class="no-dragons"
              >
                 No Dragons Reserved!
              </p>
              <div
                class="dragon-info"
              />
            </div>
          </div>
        </div>
        <div
          class="rocket-container"
        >
          <h2
            class="rocket-title "
          >
            Rockets
          </h2>
          <p
            class="no-dragons"
          >
             No Rockets Reserved Yet
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-container"
    >
      <div
        class="dragon-container"
      >
        <hr
          class="line"
        />
        <div
          class="myprofile-section"
        >
          <div
            class="dragons-section"
          >
            <h3
              class="mp-dragon-title"
            >
              Dragons
            </h3>
            <p
              class="no-dragons"
            >
               No Dragons Reserved!
            </p>
            <div
              class="dragon-info"
            />
          </div>
        </div>
      </div>
      <div
        class="rocket-container"
      >
        <h2
          class="rocket-title "
        >
          Rockets
        </h2>
        <p
          class="no-dragons"
        >
           No Rockets Reserved Yet
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
